src/App.tsx

```diff
@@ -1,14 +1,62 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+
+function prepareVisibleTodos(todos: Todo[], query: string, filterBy: string) {
+  let visibleTodos: Todo[] = [...todos];
+
+  if (query !== '') {
+    visibleTodos = visibleTodos.filter(item =>
+      item.title.toLowerCase().includes(query.toLowerCase()),
+    );
+  }
+
+  if (filterBy !== 'all') {
+    visibleTodos = visibleTodos.filter(item => {
+      switch (filterBy) {
+        case 'active':
+          return !item.completed;
+        case 'completed':
+          return item.completed;
+        default:
+          return item;
+      }
+    });
+  }
+
+  return visibleTodos;
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState<string>('');
+  const [filterBy, setFilterBy] = useState<Status>(Status.All);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  let visibleTodos = prepareVisibleTodos(todos, query, filterBy);
+
+  visibleTodos = prepareVisibleTodos(visibleTodos, query, filterBy);
+
+  const onSelect = (todo: Todo) => {
```

Comment:
```suggestion
  const handleSelect = (todo: Todo) => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364